docs/source/howto/data.rst

```diff
@@ -260,6 +260,338 @@ Finding data
+
+The data that are organized in the groups can now be accessed separately.
+Let's make a prototype script that would allow us to submit calculations for the structures belonging to the ``promising_structures`` group.
+For that we need to make a query:
+
+.. code-block:: python
+
+    q = QueryBuilder()
+    q.append(Group, filters={'label':'promising_structures'}, tag='group')
+    q.append(StructureData, with_group='group')
+
+The results of this query can be used, for instance, for subsequent simulations:
+
+.. code-block:: python
+
+    for structure in q.all(flat=True):
+        builder = SomeWorkChain.get_builder()
+        builder.structure = structure
+        ...
+        submit(builder)
+
+Note, however, that one can also use ``group.nodes`` to access the nodes of the group.
+To achieve the same result ab above one would need to do something as follows:
+
+.. code-block:: python
+
+    group = Group.get(label='promising_structures')
+    # Here make sure to include only structures, as group can contain any nodes.
+    structures = [s for s in group.nodes if isinstance(nodes, StructureData)]
+    for structure in structures:
+        builder = SomeWorkChain.get_builder()
+        builder.structure = structure
+        ...
+        submit(builder)
+
+
+For the postprocessing of the results, one can also make use of the groups.
+To find the structure that has a property ``property_a`` lower than ``1`` and at the same time belongs to the ``promising_structures`` group, one would need to build a query that looks as follows:
+
+.. code-block:: python
+
+    q = QueryBuilder()
+    q.append(Group, filters={'label':'promising_structures'}, tag='group')
+    q.append(StructureData, with_group='group', tag='structure', project='*')
+    q.append(SomeWorkChain, with_incoming='structure', tag='calculation')
+    q.append(Dict, with_incoming='calculation', filters={'attributes.property_a':{'<':1}})
+
+As a result, ``q.all(flat=True)`` would contain all the structures matching the criteria.
+
+Case III: use groups for data exporting
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Groups can be very handy for selecting the parts of the database for sharing.
+Suppose, you want to share all the simulations you've done for the structures in the ``promising_structures`` group.
+
+.. code-block:: console
+
+    $ verdi export create export.aiida -G promising_structures
+    STARTING EXPORT...
+    RETRIEVING NODES FROM GROUPS...
+    RETRIEVING LINKED NODES AND STORING LINKS...
+    STORING DATABASE ENTRIES...
+    Exporting a total of 1461 db entries, of which 988 nodes.
+    STORING NODE ATTRIBUTES AND EXTRAS...
+    STORING GROUP ELEMENTS...
+    STORING DATA...
+    STORING REPOSITORY FILES...
+    File written in       22.5 s.
+    Success: wrote the export archive file to export.aiida
+
+Keep in mind that such an export operation would not only export the structures that are belonging to the group, but also the nodes linked to them.
```

Comment:
```suggestion
Keep in mind that such an export operation would not only export the structures that are part of the group, but also the nodes linked to them.
```

Reply:
Also add a reference to this section here: https://aiida.readthedocs.io/projects/aiida-core/en/docs-revamp/topics/provenance/consistency.html#traversal-rules